zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -9305,6 +9320,51 @@ components:
 
             For now, however, the start instruction is implicitly a "startBeforeElement" instruction
           type: string
+    RuntimeInstructionType:
+      type: string
+      description: |
+        The type of the runtime instruction. For now, only SUSPEND_PROCESS_INSTANCE is supported.
+
+        Caution: process instance suspension is an alpha feature and may be subject to change
+        in future releases.
+
+        Note the following limitations of suspended process instances:
+          - Suspended process instances currently cannot be resumed. The only possible state
+            transition is to cancel the suspended instance.
+          - Process instances suspended with the current implementation may not be fully recoverable
+            in the future, when the resume functionality is implemented.
+          - Suspended process instances currently cannot be modified or migrated.
+
+        It is therefore not recommended to use process instance suspension outside of testing or
+        debugging context.
+      enum:
+        - SUSPEND_PROCESS_INSTANCE
+    ProcessInstanceCreationRuntimeInstruction:
+      type: object
+      required:
+        - type
+      properties:
+        type:
+          $ref: "#/components/schemas/RuntimeInstructionType"
+      discriminator:
+        propertyName: type
+        mapping:
```

Comment:
🔧 I'm not sure whether we need the descriminator. @tmetzke shared this with me recently: https://bump.sh/blog/the-discriminator-in-openapi-is-generally-redundant-and-confusing/

Reply:
I don't see any good way to implement this without `const`, here's what I considered:
- a simple string without const does not define adequate limitations
- an enum field in the parent object (`ProcessInstanceCreationRuntimeInstruction`) without the discriminator does not define any mapping between values and specific types
- an in-place enum definition in specific implementations (e.g. `SuspensionInstruction` defines an enum with only `SUSPEND_PROCESS_INSTANCE`) will lead to multiple enum classes created during code generation

Reply:
I believe we can leave the discriminator out for now; other APIs define different alternative types and don't need this either. But I might be missing something, or the use case here might be more specific, so it's needed.

Reply:
Actually, on re-review, I think there's a good use for the type discriminator here. It helps the user select the runtime instruction to apply. Let's keep it.